### ISSUE TITLE
Use the new %pypi_source macro for Source URL

### ISSUE THREAD

**Issue by hroncok** [fedora-python/pyp2rpm#172]
See: https://pagure.io/packaging-committee/issue/759#comment-517444

Example:

```spec
%global pypi_name Jinja2

Name:           python-%{pypi_name}
Version:        2.8
...

Source0:        %pypi_source

```

**Comment by gordonmessmer** [comment]
Is that just a matter of replacing the Source0 line in some of the templates, or are there conditions under which %pypi_source wouldn't be correct for some packages on a platform that provides that macro?  

It seems like we might want to check the URL extracted in pypi_metadata_extension to see if it ends in tar.gz and provide the macro alone if it matches, and the macro with arguments if it doesn't.

**Comment by hroncok** [comment]
> Is that just a matter of replacing the Source0 line in some of the templates

I think so, yes.

Also, we should only do this if it actually matches a pattern. That is not always true for packages with non-canonical names (such as dashes vs. underscores).

**Comment by gordonmessmer** [comment]
Resolved in v3.3.4
